File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.js
class RandomizedSet:

    def __init__(self):
        self.mapp = {}
        self.count = 0

    def insert(self, val: int) -> bool:
        if self.mapp.get(val) is None:
            self.mapp[val] = 1
            return True
        return False 

    def remove(self, val: int) -> bool:
        if self.mapp.get(val) is not None:
            self.mapp.pop(val)
            return True
        return False

    def getRandom(self) -> int:
        ran = [i for i in self.mapp.keys()]
        return random.choice(ran)


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.cpp
class RandomizedSet {
public:
    vector<int>arr;
    unordered_map<int, int> mp;
    RandomizedSet() { }
    
    bool insert(int val) {   
        auto it=mp.find(val);
        if(it == mp.end()){
            arr.push_back(val);
            mp[val]=arr.size()-1;
            return true;
        }
        return false;
    }
    bool remove(int val) {
        auto it=mp.find(val);
        if(it != mp.end()){
            arr[it->second]=arr.back();
            arr.pop_back();
            mp[arr[it->second]]=it->second;
            mp.erase(val);
            return true;
        }
        return false;
    }
    int getRandom() {
        return arr[rand()%arr.size()];
    }
};
